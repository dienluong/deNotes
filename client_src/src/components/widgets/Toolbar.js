import React, { Fragment } from 'react';
import Tool from './Tool';

function Toolbar({ toolsMap }) {
  const toolsSet = [];
  for (let [label, handler] of toolsMap) {
    toolsSet.push(<Tool key={label} label={label} onClick={ handler }/>);
  }
  return (
    <Fragment>
      { toolsSet }
    </Fragment>
  );
}

export default Toolbar;

// TODO: add proptypes
