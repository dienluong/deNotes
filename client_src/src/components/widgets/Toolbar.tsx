import React from 'react';
import styles from './Toolbar.module.css';
import Tool from './Tool';

// Types
type PropsT = {
  toolsMap: Map<{label: string, icon: JSX.Element}, (...args: any) => any>;
}

function Toolbar({ toolsMap }: PropsT) {
  const toolsSet = [];
  for (let [assets, handler] of toolsMap) {
    toolsSet.push(<Tool key={ assets.label } label={ assets.label } onClick={ handler }>{ assets.icon }</Tool>);
  }
  return (
    <div className={ styles.dnt__toolbar } >
      { toolsSet }
    </div>
  );
}

export default Toolbar;
