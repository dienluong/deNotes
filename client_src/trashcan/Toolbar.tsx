import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import MuiToolbar from '@material-ui/core/Toolbar';
import styles from './Toolbar.module.css';
import Tool from './Tool';

// Types
type PropsT = {
  toolsMap: Map<{ label: string, icon: React.ReactNode }, (...args: any) => any>;
  className: string;
  [key: string]: any;
}

function Toolbar({ toolsMap, className, ...otherProps }: PropsT) {
  const toolsSet = [];
  for (let [assets, handler] of toolsMap) {
    toolsSet.push(<Tool key={ assets.label } label={ assets.label } onClick={ handler }>{ assets.icon }</Tool>);
  }
  return (
    <AppBar { ...otherProps }>
      <MuiToolbar className={ `${className} ${styles.dnt__toolbar}` } >
        { toolsSet }
      </MuiToolbar>
    </AppBar>
  );
}

export default Toolbar;
