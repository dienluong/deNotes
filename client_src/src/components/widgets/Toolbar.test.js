import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import 'jest-dom/extend-expect';
import Toolbar from './Toolbar';

afterEach(cleanup);

it('renders a group of tools from list of labels and handlers received', () => {
  const tools = new Map();
  const numButtons = 3;
  const labelPrefix = 'tool';

  for (let i = 1; i <= numButtons; i += 1) {
    tools.set(`${labelPrefix} ${i}`, jest.fn());
  }
  const { queryAllByText } = render(<Toolbar toolsMap={ tools } />);

  tools.forEach((ignore, key) => expect(queryAllByText(key)).toHaveLength(1));
  const buttons = queryAllByText(labelPrefix, { exact: false });
  expect(buttons).toHaveLength(3);
  buttons.forEach(b => expect(b).toBeVisible());
  buttons.forEach(b => fireEvent.click(b));
  tools.forEach(handler => expect(handler).toBeCalledTimes(1));
});
