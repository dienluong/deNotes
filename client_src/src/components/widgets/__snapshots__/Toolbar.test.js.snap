// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a group of tools from list of labels and handlers received 1`] = `
<div>
  <button>
    tool 1
  </button>
  <button>
    tool 2
  </button>
  <button>
    tool 3
  </button>
</div>
`;
